lib/src/controls/flyouts/menu.dart

```diff
@@ -0,0 +1,532 @@
+  ///
+  /// ![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/contextmenu_rs2_icons.png)
+  ///
+  /// Usually an [Icon] widget,
+  final Widget? leading;
+  final Widget text;
+  final Widget? trailing;
+  final VoidCallback? onPressed;
+  final bool selected;
+
+  bool _useIconPlaceholder = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final size = ContentSizeInfo.of(context).size;
+    return Container(
+      width: size.isEmpty ? null : size.width,
+      padding: MenuFlyout.itemsPadding,
+      child: FlyoutListTile(
+        selected: selected,
+        showSelectedIndicator: false,
+        icon: leading ??
+            () {
+              if (_useIconPlaceholder) return const Icon(null);
+              return null;
+            }(),
+        text: text,
+        trailing: IconTheme.merge(
+          data: const IconThemeData(size: 12.0),
+          child: trailing ?? const SizedBox.shrink(),
+        ),
+        onPressed: onPressed,
+      ),
+    );
+  }
+}
+
+/// Represents a horizontal line that separates items in a [MenuFlyout].
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+class MenuFlyoutSeparator extends MenuFlyoutItemBase {
+  /// Creates a menu flyout separator
+  const MenuFlyoutSeparator({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final size = ContentSizeInfo.of(context).size;
+
+    return SizedBox(
+      width: size.width,
+      child: const Padding(
+        padding: EdgeInsetsDirectional.only(bottom: 5.0),
+        child: Divider(
+          style: DividerThemeData(horizontalMargin: EdgeInsets.zero),
+        ),
+      ),
+    );
+  }
+}
+
+enum SubItemShowBehavior {
+  /// Whether the sub-menu will be shown on item press
+  press,
+
+  /// Whether the sub-menu will be shown on item hover
+  ///
+  /// This is the default behavior.
+  hover,
+}
+
+/// Represents a menu item that displays a sub-menu in a [MenuFlyout].
+///
+/// ![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/menu-bar-submenu.png)
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+class MenuFlyoutSubItem extends MenuFlyoutItem {
+  /// Creates a menu flyout sub item
+  MenuFlyoutSubItem({
+    super.key,
+    super.leading,
+    required super.text,
+    super.trailing = const Icon(FluentIcons.chevron_right),
+    required this.items,
+    this.showBehavior = SubItemShowBehavior.hover,
+    this.showHoverDelay = const Duration(milliseconds: 600),
+  }) : super(onPressed: null);
+
+  GlobalKey<__MenuFlyoutSubItemState>? _key;
```

Comment:
Could you explain this?

Reply:
It is the key of `_MenuFlyoutSubItem`, built in the `build` method. It can not use the parent `key` widget because it's already used by `MenuFlyout` to build the widget. It is assigned by `MenuFlyout` with a key generated by `generateKeys()`. It is used to close the child menu when its parent is closed